cmd/kubemark/hollow-node.go

```diff
@@ -145,11 +159,75 @@ func run(config *hollowNodeConfig) {
 	}
 
 	// create a client to communicate with API server.
-	clientConfig, err := config.createClientConfigFromFile()
-	if err != nil {
-		klog.Fatalf("Failed to create a ClientConfig: %v. Exiting.", err)
+	var clientConfig *restclient.Config
+	var err error
+	if config.AuthProvider == "" {
+		clientConfig, err = config.createClientConfigFromFile()
+		if err != nil {
+			klog.Fatalf("Failed to create a ClientConfig: %v. Exiting.", err)
+		}
+	} else {
+		switch config.AuthProvider {
+		case "eks":
+			if config.ClusterContextName == "" {
+				klog.Fatalf("empty cluster context name is given for auth provider %q", config.AuthProvider)
+			}
+			if config.Region == "" {
+				klog.Fatalf("empty region is given for auth provider %q", config.AuthProvider)
+			}
+			if config.ClusterName == "" {
+				klog.Fatalf("empty cluster name is given for auth provider %q", config.AuthProvider)
+			}
+
+			var kcfg *clientcmdapi.Config
+			kcfg, err = clientcmd.LoadFromFile(config.KubeconfigPath)
+			if err != nil {
+				klog.Fatalf("error while loading kubeconfig from file %q: %v", config.KubeconfigPath, err)
+			}
+			v, ok := kcfg.Clusters[config.ClusterContextName]
+			if !ok {
+				keys := make([]string, 0, len(kcfg.Clusters))
+				for k := range kcfg.Clusters {
+					keys = append(keys, k)
+				}
+				if err != nil {
+					klog.Fatalf("cluster context name %q not found in kubeconfig 'clusters' (%v)", config.ClusterContextName, keys)
+				}
+			}
+
+			klog.Infof("setting rest client to server %q from kubeconfig %q", v.Server, config.KubeconfigPath)
+			clientConfig = &restclient.Config{
+				Host: v.Server,
+				TLSClientConfig: restclient.TLSClientConfig{
+					// no need base64-decode, the field is already PEM-encoded when returned from 'clientcmd.LoadFromFile'
+					CAData: v.CertificateAuthorityData,
+				},
+				AuthProvider: &clientcmdapi.AuthProviderConfig{
+					Name: config.AuthProvider,
+					Config: map[string]string{
+						// TODO: support temporary credentials
+						"region":       config.Region,
+						"cluster-name": config.ClusterName,
+					},
+				},
+			}
+
+			restclient.RegisterAuthProviderPlugin(config.AuthProvider, newEKSAuthProvider)
+			klog.Infof("successfully registered auth provider %q with region %q and cluster name %q",
+				config.AuthProvider,
+				config.Region,
+				config.ClusterName,
+			)
+
+		default:
+			klog.Fatalf("unknown auth provider %q given", config.AuthProvider)
+		}
 	}
 
+	clientConfig.ContentType = config.ContentType
+	clientConfig.QPS = 10
+	clientConfig.Burst = 20
```

Comment:
@shyamjvs I removed these lines from `createClientConfigFromFile` and put them here, to avoid redundant `clientConfig` updates.

Reply:
Ok, got it. Can you instead just move this auth-provider logic into `createClientConfigFromFile` function as that's where the client-config initiation should be done IMO?